tests/lib/fde-setup-hook/fde-setup.go

```diff
@@ -42,6 +68,21 @@ func runFdeSetup() error {
 		return err
 	}
 
+	var jsStrict fdeSetupJSONStrictBase64
+	if err := json.Unmarshal(output, &jsStrict); err != nil {
+		return err
+	}
+
+	// verify that the two de-coding mechanisms agree on the key, manually
+	// decoding the base64 string in the stricter case
+	decodedBase64Key, err := base64.StdEncoding.DecodeString(jsStrict.Key)
+	if err != nil {
+		return fmt.Errorf("fde-setup-request is not valid base64: %v", err)
+	}
+	if err := byteSlicesEqual(decodedBase64Key, js.Key); err != nil {
```

Comment:
I was wondering if "bytes.Equal()" and simplifying the json struct would work here too? Something like:
```
--- a/tests/lib/fde-setup-hook/fde-setup.go
+++ b/tests/lib/fde-setup-hook/fde-setup.go
@@ -37,12 +37,7 @@ type fdeSetupJSON struct {
 // as a string, which _must_ be a base64 encoded version of the same []byte Key
 // we have above, the handler below validates this as a test
 type fdeSetupJSONStrictBase64 struct {
-       Op string `json:"op"`
-
-       Key     string `json:"key,omitempty"`
-       KeyName string `json:"key-name,omitempty"`
-
-       Models []map[string]string `json:"models,omitempty"`
+       Key string `json:"key,omitempty"`
 }
 
 func byteSlicesEqual(b1, b2 []byte) error {
@@ -79,8 +74,8 @@ func runFdeSetup() error {
        if err != nil {
                return fmt.Errorf("fde-setup-request is not valid base64: %v", err)
        }
-       if err := byteSlicesEqual(decodedBase64Key, js.Key); err != nil {
-               return fmt.Errorf("fde-setup-request is not strictly the same base64 decoded as binary decoded: %v", err)
+       if !bytes.Equal(decodedBase64Key, js.Key) {
+               return fmt.Errorf("fde-setup-request is not strictly the same base64 decoded as binary decoded: %v != %v", decodedBase64Key, js.Key)
        }
 
        var fdeSetupResult []byte
```
It seems to not break the tests at least.

Reply:
done in https://github.com/snapcore/snapd/pull/10072